src/Context/AnnotatedClassDiscovery.php

```diff
@@ -11,7 +11,13 @@
 use \Drupal\Core\Plugin\Discovery\AnnotatedClassDiscovery as CoreAnnotatedClassDiscovery;
 
 /**
- * Extends the annotation class discovery for usage with Rules context.
+ * Extends the annotation class discovery for usage with our modified annotations.
+ *
+ * We modify the annotations classes for ContextDefinition and for Condition.
+ * This class makes sure that our plugin managers apply our versions of those
+ * of Core.
+ *
+ * @todo This may need to move into a different directory.
```

Comment:
Whereto? Let's just either move it here or  not - but not add a todo for something as simple as that.